File: cs/competitive_programming/leetcode_fulltimes/2022/november/1_two_sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        store = {}
        for i in range( len(nums) ):
            store[nums[i]] = i

        for i in range( len(nums) ):
            to_find = target-nums[i]
            if to_find in store and store[to_find] != i:
                return [i,store[to_find]]

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n = len(nums)
        store = {}

        for i in range(n):
            store[nums[i]]=i

        for i in range(n):
            to_find = target-nums[i]
            if (to_find in store) and i!=store[to_find]:
                return [i, store[to_find]]

File: cs/competitive_programming/leetcode_fulltimes/2023/januarary/cpp/1_two_sum.cpp
#include <iostream>
#include <vector>
#include <map>

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {

        unordered_map<int, int> visited;

        for(int i=0; i<nums.size(); i++){
            int n = nums[i];
            int complement = target-nums[i];

            if(visited.count(complement)){
                return {i, visited[complement]};
            }

            visited[nums[i]]=i;

        }
        return {};
    }
};


class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        int n = nums.size();
        map<int, int> mp;


        for(int i=0; i<n; i++){
            mp[nums[i]]=i;
        }


        for(int i=0; i<n; i++){
            int complement=target-nums[i];

            if(mp.count(complement) && (i!=mp[complement])){
                return {i, mp[complement]};
            }

        }

        return {};
    }
};